src/engraving/libmscore/textedit.cpp

```diff
@@ -248,12 +248,18 @@ void TextBase::insertText(EditData& ed, const QString& s)
 bool TextBase::edit(EditData& ed)
 {
     TextEditData* ted = static_cast<TextEditData*>(ed.getData(this));
+    if (!ted) {
+        return false; 
+    }
     TextCursor* cursor = ted->cursor();
 
     // do nothing on Shift, it messes up IME on Windows. See #64046
     if (ed.key == Qt::Key_Shift) {
         return false;
     }
+    if (ed.key == Qt::Key_Escape) {
+        return false;
```

Comment:
Seems MU3 was behaving quite differently here, the Text elements were only getting keystrokes depending on what widget group and its related "ActionGroup" was active, but that concept no longer exists in MU4, so all keystrokes are arriving here, and the element must return false for any it doesn't want.